src/confluent_kafka/src/confluent_kafka.c

```diff
@@ -1620,7 +1620,10 @@ PyObject *c_Node_to_py(const rd_kafka_Node_t *c_node) {
 
         cfl_PyDict_SetInt(kwargs, "id", rd_kafka_Node_id(c_node));
         cfl_PyDict_SetInt(kwargs, "port", rd_kafka_Node_port(c_node));
-        cfl_PyDict_SetString(kwargs, "host", rd_kafka_Node_host(c_node));
+        if (rd_kafka_Node_host(c_node))
```

Comment:
Now I have a doubt that the fix should be here in CKPy, because this uses the Metadata response too for the replicas and isrs and it uses the Metadata host here (not nullable)
https://github.com/apache/kafka/blob/e8cd661becb08279702f1670bb8f1d816537adea/clients/src/main/resources/common/message/MetadataResponse.json#L54
And broker metadata must be present for all the indexed brokers

Reply:
OK, here it can be None because it's possible not all the brokers are returned from the Metadata call. And in that case it just sets the broker id here
https://github.com/confluentinc/librdkafka/blob/c95d2ee950bda779c63843d376aab1d90ed8dfa9/src/rdkafka_aux.c#L281

In the `list_topics` API, instead, only the broker id list is returned in the `replicas` and `isrs` fields so the broker list in `ClusterMetadata` only contains complete `BrokerMetadata` objects.

Reply:
It seems it's the expected behavior. I've created a PR for changing the documentation [here](https://github.com/apache/kafka/pull/15809).
An update: in Go and .NET it doesn't happen because it either uses `C.GoString` or `PtrToStringUTF8` that returns `nil` or `null`.